src/robots/msj_platform.cpp

```diff
@@ -157,11 +159,52 @@ class MsjPlatform: public cardsflow::kindyn::Robot{
 
     }
 
+    ///find the closest limit point when the robot is in infeasible state
+    VectorXd findClosestJointLimit(double q0, double q1, double q3){
+        ROS_INFO("FINDING CLOSEST JOINT LIMIT");
+        VectorXd closestLimit;
+        closestLimit.resize(number_of_dofs);
+        closestLimit.setZero();
+        double distance = 0.0, smallestDistance = numeric_limits<double>::max();
```

Comment:
do you need to define distance in advance, or can you define it where you set it?

Reply:
it can be